src/js/core/UpdateStream.js

```diff
@@ -0,0 +1,100 @@
+import { Observable } from "rxjs/Observable";
+import UserSettingsStore from "#SRC/js/stores/UserSettingsStore";
+import { request } from "@dcos/http-service";
+import Util from "#SRC/js/utils/Util";
+import { SAVED_STATE_KEY } from "#SRC/js/constants/UserSettings";
+import compareVersions from "compare-versions";
+
+const CHECK_DELAY = 24 * 60 * 60 * 1000; // Once every 24 hours
+const DCOS_UI_VERSION = "dcosUIVersion";
+
+export const UpdateStreamType = Symbol("UpdateStreamType");
+
+function getFromLocalStorage(key) {
+  const value = Util.findNestedPropertyInObject(
+    UserSettingsStore.getKey(SAVED_STATE_KEY),
+    DCOS_UI_VERSION + "." + key
+  );
+
+  return value;
+}
+
+function setInLocalStorage(key, value) {
+  const savedStates = UserSettingsStore.getKey(SAVED_STATE_KEY) || {};
+  savedStates[DCOS_UI_VERSION] = {
+    dismissedVersion: getFromLocalStorage("dismissedVersion") || null,
+    lastTimeCheck: getFromLocalStorage("lastTimeCheck") || null
+  };
+  savedStates[DCOS_UI_VERSION][key] = value;
+  UserSettingsStore.setKey(SAVED_STATE_KEY, savedStates);
+}
+
+function getContentType({ action, actionType, entity, version }) {
+  return `application/vnd.dcos.${entity}.${action}-${actionType}+json;charset=utf-8;version=${version}`;
+}
+
+function showNotification(newVersion) {
+  // TODO: show notification
+  // When dismissing, save dismissed version in local storage
+  const currentVersion = window.DCOS_UI_VERSION;
+  console.log(
+    "A new version of DCOS UI is available. Upgrade " +
+      currentVersion +
+      " to " +
+      newVersion +
+      ". Dismiss."
+  );
+}
+
+setInLocalStorage("dismissedVersion", "2.24.2"); // stub
+const dismissedVersion = Observable.of(
+  getFromLocalStorage("dismissedVersion") || "0"
+);
+
+const fetchedVersion = request("/package/list-versions", {
+  method: "POST",
+
+  headers: {
+    "Content-Type": getContentType({
+      action: "list-versions",
+      actionType: "request",
+      entity: "package",
+      version: "v1"
+    }),
+    Accept: getContentType({
+      action: "list-versions",
+      actionType: "response",
+      entity: "package",
+      version: "v1"
+    })
+  },
+  body: JSON.stringify({ includePackageVersions: true, packageName: "dcos-ui" })
+}).retry(4);
+
+export const compare = Observable.timer(0, CHECK_DELAY).switchMap(() =>
+  Observable.combineLatest(
+    dismissedVersion,
+    fetchedVersion,
+    (dismissedVersion, newVersionObject) => {
+      if (
+        new Date().getTime() <
```

Comment:
Why do we need this conditional?

Reply:
It's something @brandonc suggested and I liked the idea.

Reply:
So we don't re-check if a user refreshes the browser?

Reply:
No, because I think otherwise the 24-hour timer would be sort of useless, I don't think anyone will keep the UI 24 hours without restarting/refreshing. Do you think that we should?

Reply:
Sentry suggests that some folks keep the UI open over weekends

Reply:
Regardless I'd suggest rewriting this check from being imperative to an appropriate rxjs operator like `filter`